File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		List<Integer> dividor = new ArrayList<Integer>();
		double a = Math.sqrt(2*N);
		int b = (int) a;
		int c;
		int ans =0;
		if((a-b)==0) {
			c = b;
		}else {
			c = b+1;
		}
		for(int i = c; i<=2*N; i++) {
			if(2*N%i==0) {
				dividor.add(i);
			}
			
		}
		
		for(int i = 0; i<dividor.size(); i++) {
			
			double p = (double) (dividor.get(i)+2*N/dividor.get(i)-1)/2;
			
			int q = (int) p;
			if((q-p)==0) {
				ans++;
			}
		
		}
		System.out.println(ans);
	}

}

File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.py
import math
N = int(input())
count = 0
# print(int(math.sqrt(N)))
for i in range(1, int(math.sqrt(2*N))+1):
    if (2*N)%i ==0:
        if (i % 2 == 0 and (int((2*N)/i))%2 ==1) or (i % 2 == 1 and (int((2*N)/i))%2 ==0):
            #print(i)
            count+=1
print(int(count))